pkgs/applications/science/math/mathematica/default.nix

```diff
@@ -93,6 +94,12 @@ stdenv.mkDerivation rec {
 
     echo "=== Running MathInstaller ==="
     ./MathInstaller -auto -createdir=y -execdir=$out/bin -targetdir=$out/libexec/Mathematica -silent
+
+    # Fix library paths
```

Comment:
Can you use `ldd` on all executables and check which ones don't have a correct reference for `zlib`.
The rpath is fixed for executables in `libexec`, it may be that the executables you modify here need `zlib` instead.